### ISSUE TITLE
Thin doesn't work properly

### ISSUE THREAD

**Issue by bbbales2** [stan-dev/cmdstanr#114]
**Describe the bug**
Thin doesn't seem to be working correctly.

I requested 1000 post-warmup draws per chain with thin = 5 and when I call draws I get:
```
> fit1$draws() %>% dim
[1] 1000    4    7
Warning message:
In rstan::read_stan_csv(self$output_files()) :
  the number of iterations after warmup found (1000) does not match iter/warmup/thin from CSV comments (200,200,200,200)
```

**To Reproduce**
This is the model I was using:
```
data {
  int N; //number obs
  int K; //number of columns in the model matrix
  int y[N]; //response
  matrix[N,K] X; 
}

parameters {
  real alpha;
  vector[K] beta; //the regression parameters
}
model {  
  alpha ~ normal(0,1.5);
  
  beta ~ normal(0,1);
  
  y ~ poisson_log(alpha + X * beta); //version 1
} 
```

And the R code:

```
model1 = cmdstan_model("poisson_log1.stan", quiet = FALSE)

N = 20
K = 5
alpha = rnorm(1, 0, 1)
beta = rnorm(K, 0, 1)

X = matrix(rnorm(N * K), nrow = N)
y = rpois(N, exp(alpha + X %*% beta))

data = list(N = N,
            K = K,
            X = X,
            y = y)

fit1 = model1$sample(data = data, num_cores = 4, thin = 5)
```

**Expected behavior**
The array should have been of size 200x4x7

**Operating system**
Ubuntu 18.04

**CmdStanR version number**
‘0.0.0.9000’

**Comment by rok-cesnovar** [comment]
Thanks @bbbales2 

I am 99% that this will be fixed with #95

**Comment by rok-cesnovar** [comment]
Ran this after #95 was merged and got:

```
> dim(fit1$draws())
[1] 200   4   7
```
so closing.
